### PR TITLE
Release v4.0.1

### PR DESCRIPTION
**Pull request by asifamingov**

### CODE REVIEW

File: src/assets/_project/_blocks/utils/qg-load-google-api.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ import keys from '../data/qg-google-keys';
   // check if the hostname contains a specific word and assign the key accordingly
   if (window.location.hostname.search(/\bgithub\b/) !== -1) {
     googleApiKey = keys.defGoogle.docs;
-  } else if (window.location.hostname.search(/\bdev\b|\btest\b|\blocalhost\b/) !== -1) {
+  } else if (window.location.hostname.search(/dev|test|localhost/) !== -1) {
```
Comment (duttonw) [inline src/assets/_project/_blocks/utils/qg-load-google-api.js:16]: should we also include uat? (like recaptcha key swap?)

**Review by duttonw — APPROVED** [review]
some comments for next release, over all good :+1:

File: .circleci/config.yml [hunk 2]
```diff
@@ -31,12 +31,6 @@ jobs:
           name: Local Server
           command: npx gulp serve
           background: true
-
-      - run:
```
[the remark this answers is not in the document]
Comment (asifamingov) [reply .circleci/config.yml:35]: Need to update some of the tests with the new header and footer they are not running.

File: src/assets/_project/_blocks/components/_cards.scss [hunk 3]
```diff
@@ -32,7 +32,7 @@
       }
       .details{
         padding: 1em;
-        margin-bottom: 4em;
+        //margin-bottom: 4em;
```
[the remark this answers is not in the document]
Comment (asifamingov) [reply src/assets/_project/_blocks/components/_cards.scss:35]: made the change will send a pull request for the release 4.0.2 patch release which I just did

File: src/assets/_project/_blocks/utils/qg-load-google-api.js (hunk 1 above)
Comment (asifamingov) [reply src/assets/_project/_blocks/utils/qg-load-google-api.js:16]: made the change will send a pull request for the release 4.0.2 patch release which I just did

Comment (asifamingov) [reply src/assets/_project/_blocks/utils/qg-load-google-api.js:16]: UAT env we usually test with PROD set up to make sure it is working before PROD deployment

File: gulp/gulp-config.js [hunk 4]
```diff
@@ -4,10 +4,10 @@ const path        = require('path');
 const pjson       = require('../package.json');
 
 module.exports = {
-  versionName: 'v3.1',
+  versionName: 'v4',
   version: {
-    major: 3,
-    minor: 1,
+    major: 4,
+    minor: 0,
```
[the remark this answers is not in the document]
Comment (asifamingov) [reply gulp/gulp-config.js:10]: I think this is not in use, I will delete this.
I switched to use **package.json** for all versioning stuff
`  "version": "4.0.2",
  "subVersion": "4.0.2",
  "wt-version": "4.0.2",`

**Comment by ThrawnCA** [comment]
This pull request is huge and I'm not a JavaScript specialist. I'm not sure I can meaningfully review it.

**Comment by duttonw** [comment]
the problem here is that master was not updated back from release-X branches and is way out of date and is being caught up. there is no real need to review everything.

**Pull request merged**